Code/6-wire-version/src/main.cpp

```diff
@@ -167,7 +244,114 @@ void sendMQTT(){
 
 
 
-void startOTA() { // Start the OTA service
+/**
+ * Start a Wi-Fi access point, and try to connect to some given access points.
+ * Then wait for either an AP or STA connection
+ */
+void startWiFi()
+{
+  //WiFi.mode(WIFI_STA);
+  WiFi.setAutoReconnect(true);
+  WiFi.persistent(true);
+  WiFi.hostname(netHostname);
+
+  if (enableStaticIp4)
+  {
+    Serial.println("WiFi > using static IP \"" + ip4Address.toString() + "\" on gateway \"" + ip4Gateway.toString() + "\"");
+    WiFi.config(ip4Address, ip4Gateway, ip4Subnet, ip4DnsPrimary, ip4DnsSecondary);
+  }
+
+  if (enableAp)
+  {
+    Serial.println("WiFi > using WiFi configuration with SSID \"" + apSsid + "\"");
+
+    WiFi.begin(apSsid, apPwd);
+
+    Serial.print("WiFi > Trying to connect ...");
+    int maxTries = 10;
+    int tryCount = 0;
+
+    while (WiFi.status() != WL_CONNECTED)
+    {
+      delay(1000);
+      Serial.print(".");
+      tryCount++;
+
+      if (tryCount >= maxTries)
+      {
+        Serial.println("");
+        Serial.println("WiFi > NOT connected!");
+        if (enableWmApFallback)
```

Comment:
i wanted to have that fallback so people who cannot flash their own config are able to modify the network if something has changed. e.g. new router/network. all they have to do is connecting to the auto portal and set their credentials.
people like @hubaksis can just disable that to continuously retrying to get the configured network running.
maybe it's worth to put that into the wifi config web. just a checkbox where you can set that when having problems (also the behaviour can be explained there)

Reply:
@visualapproach what do you think?

Reply:
btw. i had 4 test cases.

case 1 (fallback feature **on**)
1. clean startup (without creds)
2. set creds via wifi manager > reboot
3. startup with creds
4. connection established (writes into wifi.json)
5. ~deny wifi (disable mac address)
6. monitor ..
7. ~allow wifi
8. = reconnected

case 2 (fallback feature **on**)
1. ~deny wifi 
2. startup with creds (from wifi.json)
3. pump should start; wifi not
4. wifi manager starts AP mode
5. ~allow wifi
6. enter creds again -> restart -> blabla
7. = reconnected

case 3 (fallback feature **off**)
`test steps like the first test case` 

case 4 (fallback feature **off**)
1. ~deny wifi 
2. startup with creds (from wifi.json)
3. pump should start; wifi not
4. monitor ..
5. ~allow wifi
6. monitor ..
7. = reconnected

Reply:
Don't understand why I can't reply on this from VS code, but writing on github page now. I think it's okay to have fallback autoportal as an option in web GUI (default ON). And your tests seems promising. I'll do the same when I can.